src/plugins/data_source_management/public/plugin.ts

```diff
@@ -103,13 +136,13 @@ export class DataSourceManagementPlugin
           core.getStartServices,
           params,
           this.authMethodsRegistry,
-          featureFlagStatus
+          this.featureFlagStatus
```

Comment:
Transferring some conclusion from the mini bug bash with @ruanyl for checking the changes with new navigation experience:
- have to change line, otherwise the feature flag status will be undefined when navigation feature is enabled 
- we also discovered that when the new navigation is enabled, the data source section is only registered when `data_source.enabled: true` is set to `true`. This behavior isn't correct, since my previous change(https://github.com/opensearch-project/OpenSearch-Dashboards/pull/7143) has changed this behavior. right now the correct behavior should be: data source section will always be registered no matter `data_source.enabled` is enabled or not. However, since this change also touches some of the feature flag related logic, we may need to fix this issue after merge this PR in. 

(cc: @SuZhou-Joe)

Reply:
Yes, since we will have a unified data sources management, there is no need to display 2 data sources entry(1. management > data sources 2. Dashboards management > data sources) in the left navigation.

Reply:
Thanks for the confirm! We will also unregister the old datasource plugin for observability from the left menu after the merge of this feature.